### PR TITLE
Yaml output

### PR DESCRIPTION
**Pull request by williamcroberts**
Audit the tools for printf usages and replace with tpm2_tool_output. Also verify that output is in a YAML compliant format. Add tests that verify that the output can be loaded via the YAML module in python.

Fixes: #634

### CODE REVIEW

**Comment by williamcroberts** [comment]
Looks like 
```TPM2_PT_VENDOR_STRING_2:
  raw: 0x2054504D
  value: "TPM"
```
The value has a garbage byte, can't repro locally...hmmm

**Comment by williamcroberts** [comment]
Ahh need to subtract ```i``` from ```j``` when setting the initial value as the byte pointer walks forward....

File: tools/tpm2_getcap.c [hunk 1]
```diff
@@ -159,9 +159,8 @@ int sanity_check_capability_opts (void) {
```
Comment (martinezjavier) [inline tools/tpm2_getcap.c]: In the commit message:

s/cononical/canonical
s/Ie/i.e
s/worK/work

File: tools/tpm2_getcap.c [hunk 2]
```diff
@@ -159,9 +159,8 @@ int sanity_check_capability_opts (void) {
 
     size_t i;
     for (i = 0; i < CAPABILITY_MAP_COUNT; ++i) {
-        int cmp = strncmp(capability_map [i].capability_string,
-                          options.capability_string,
-                          strlen(capability_map [i].capability_string));
+        int cmp = strcmp(capability_map [i].capability_string,
```
Comment (martinezjavier) [inline tools/tpm2_getcap.c]: Remove the space between `capability_map` and `[i]`. Nice catch BTW.

File: man/tpm2_getcap.1.md [hunk 3]
```diff
@@ -58,6 +58,11 @@ command.
     * handles-saved-session:
       Display handles about saved sessions.
 
+  * **-l**, **--list**:
+    List known supported capability names. These names can be
+    supplied as the argument to the -c option. Output is in a
```
Comment (martinezjavier) [inline man/tpm2_getcap.1.md]: \*\*-c\*\* option.

File: test/system/tests/dictionarylockout.sh [hunk 4]
```diff
@@ -32,6 +32,30 @@
 #;**********************************************************************;
 ###this script use for test the implementation tpm2_dictionarylockout 
 
+function yaml_get() {
```
Comment (martinezjavier) [inline test/system/tests/dictionarylockout.sh]: Probably we should at some point move all these functions to a lib.sh or something and source it from the tests. I haven't checked, but from previous review I remember quite similar functions so there's no need for the code duplication.

File: test/system/helpers.sh [hunk 5]
```diff
@@ -99,3 +99,22 @@ hash_alg_supported() {
         fi
     done
 }
+
+#
+# Verifies that the contexts of a file path provided
+# as the first argument loads as a YAML file.
+#
+function yaml_verify() {
```
Comment (martinezjavier) [inline test/system/helpers.sh]: Nice, you are doing what I suggested :)

But I think this should be one of the first patches in the series, since other patches are duplicating this function in the tests. Also, it seems that Github is messing with the commit order, since I saw in previous patches that you were sourcing this file and using the function...

File: test/system/tests/quote.sh [hunk 6]
```diff
@@ -51,14 +51,7 @@ Handle_ak_quote=0x81010016
 Handle_ek_quote=0x81010017
```
Comment (martinezjavier) [inline test/system/tests/quote.sh]: Can you trim the quoted line in the commit message, we get the idea. No need to have such a long hex number :)

File: test/system/tests/nv.sh [hunk 7]
```diff
@@ -58,26 +60,6 @@ onerror() {
 }
 trap onerror ERR
 
-function yaml_get() {
```
Comment (martinezjavier) [inline test/system/tests/nv.sh]: This should also be in a separate patch. Basically I would do:

1) Factor out all the existing funcions in helper.sh as the first patch
2) Add any new functions that are going to be used
3) Do all the YAML conversions and sourcing helpers.sh in the tests.

File: test/system/tests/create.sh [hunk 8]
```diff
@@ -49,27 +49,6 @@ cleanup() {
 }
 trap cleanup EXIT
 
-function yaml_get() {
```
Comment (martinezjavier) [inline test/system/tests/create.sh]: Same here, I think it will be easier to read if you first factor out yaml_get(), then on a subsequent patch add yaml_get_kv() and finally on another patch change the users.

File: lib/tpm2_alg_util.c [hunk 9]
```diff
@@ -351,7 +351,7 @@ UINT8* tpm2_extract_plain_signature(UINT16 *size, TPMT_SIGNATURE *signature) {
         if (!buffer) {
             goto nomem;
         }
-        memcpy(buffer, &hmac_sig, *size);
```
Comment (martinezjavier) [inline lib/tpm2_alg_util.c]: I would also have this at the beginning of the set since is a trivial fix (although maybe you did since as mentioned Github seems to mess with the commits order).

**Comment by williamcroberts** [comment]
@martinezjavier the ordering on github drives me nuts, it uses the timestamp versus the actual commit parent order. I plan on re-writing the history on this and bring in the helper.sh changes first.
